### PR TITLE
domain/infosync: migrate test-infra to testify

### LINKED ISSUE #26155: migrate test-infra to testify for `domain/infosync` pkg

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### What problem does this PR solve?

Issue Number: close #26155 

### Release note

- No release note

### CODE REVIEW

File: domain/infosync/info_test.go [hunk 1]
```diff
@@ -20,172 +20,127 @@ import (
...
-	topo, err := info.getTopologyFromEtcd(ctx)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if topo.StartTimestamp != 1282967700000 {
-		t.Fatal("start_timestamp of topology info does not match")
-	}
-	if v, ok := topo.Labels["foo"]; !ok || v != "bar" {
-		t.Fatal("labels of topology info does not match")
```
Comment (xhebox) [inline domain/infosync/info_test.go:121]: Maybe keep these comments.

File: domain/infosync/info_test.go [hunk 2]
```diff
@@ -20,172 +20,127 @@ import (
...
+	require.NoError(t, err)
 
-	// check ttl key
-	ttlExists, err := info.ttlKeyExists(ctx)
+	ttlExists, err = info.ttlKeyExists(ctx)
+	require.NoError(t, err)
+	require.True(t, ttlExists)
+}
+
+func (is *InfoSyncer) getTopologyFromEtcd(ctx context.Context) (*topologyInfo, error) {
```
Comment (xhebox) [inline domain/infosync/info_test.go:116]: What about moving the two methods out of `_test.go`, the package is not with suffix `_test`: there should not be any priv/access problem.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

File: domain/infosync/info_test.go (hunk 1 above)
Comment (tisonkun) [reply domain/infosync/info_test.go:121]: Do you mean exactly this comment or all comments in this file? I think assertion itself provides a few information to understand what fails.

File: domain/infosync/info_test.go (hunk 2 above)
Comment (tisonkun) [reply domain/infosync/info_test.go:116]: Make senses. I'd like to move helper functions to a separated file, but don't find a good name yet. What `_test.go` it should be?

Comment (xhebox) [reply domain/infosync/info_test.go:116]: I mean moving it into `info.go`. But, yes, TiDB itself is not using that, it is for PD... I did not notice that. Then it is not necessary change request.

File: domain/infosync/info_test.go (hunk 1 above)
Comment (xhebox) [reply domain/infosync/info_test.go:121]: Makes sense and acceptable, resolved.

File: domain/infosync/info_test.go (hunk 2 above)
Comment (tisonkun) [reply domain/infosync/info_test.go:116]: Fine. I will mark this discussion as resolved then.

**Review by xhebox — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**